client/signup/steps/website-content/wordpress-media-upload.tsx

```diff
@@ -156,6 +156,13 @@ export function WordpressMediaUpload( {
 	const allowedFileTypes = mediaType === 'VIDEO' ? allowedVideoExtensions : allowedImageExtensions;
 	const allowedFileTypesString = allowedFileTypes.map( ( type ) => `.${ type }` ).join();
 
+	// Initialize uploadState if media has already been uploaded.
```

Comment:
In L150, we initialize the `uploadState`, but the `media` prop is updated after the initial render. So the `uploadState` remains as `UPLOAD_STATES.NOT_SELECTED` even though there is a valid `url`.

This hook can be thought of as the equivalent of a `getDerivedStateFromProps` method.